src/samplebuffer.h

```diff
@@ -0,0 +1,90 @@
+#ifndef SAMPLEBUFFER_H
+#define SAMPLEBUFFER_H
+
+#include "util/types.h" // CSAMPLE
+
+#include <algorithm> // std::swap
+
+// A sample buffer with properly aligned memory to enable SSE optimizations.
+//
+// No resize operation is provided intentionally for maximum efficiency!
+// If the size of an existing sample buffer needs to be altered after
+// construction this can simply be achieved by swapping the contents with
+// a temporary sample buffer that has been constructed with the appropriate
+// size:
+//
+//     SampleBuffer sampleBuffer(oldSize);
+//     ...
+//     SampleBuffer(newSize).swap(sampleBuffer);
+//     // Now the data in sampleBuffer is uninitialized and
+//     // sampleBuffer.size() == newSize
+//
+// No sample data is copied when swapping the contents of two samples buffers
+// for resizing!
+class SampleBuffer {
+    Q_DISABLE_COPY(SampleBuffer);
+
+  public:
+    typedef size_t size_type;
+    typedef CSAMPLE value_type;
```

Comment:
Since this is no template class I would prefer to explicit use CSAMPLE, otherwise we will have a lot of implicit casts which may confuse readers (including me). 
Qt uses int for all loops and size types, thats was why I have picked int when optimized the Sample functions. We should use int here too to avoid signed unsigned castings and 64 bit to 32 bit casts.